### ISSUE TITLE
Improvements to `Timestamp`

### ISSUE THREAD

**Issue by binkley** [klogging/klogging#24]
These tests illustrate shortcomings to `Timestamp`:
```
    describe("Timestamps") {
        val somewhen = Timestamp(2L, 3L)

        it("represents as UTC time") {
            somewhen.isoString shouldBe "1970-01-01T00:00:02.000000003Z"
        }
        it("represents as local time") {
            // TODO: Control for locale and time zone
            somewhen.localString shouldBe "1970-01-01 08:00:02.000000003"
        }
    }
```

1. Local time does not show the time zone or offset from GMT
2. No format configuration available
   - Some users may want to control timestamp format, _eg_, for consumption by a system with format limitations
   - Item # 1 is less a concern if users can configure format
3. Valid ISO 8601 formats include, meaning, these are the "official" formats (usually with an extension for millis, micros, or nanos), so there is not a single universal choice:
   - 2021-08-06T03:08:10+00:00
   - 2021-08-06T03:08:10Z
   - 20210806T030810Z
4. For timestamps, local formatting should be *strongly* discouraged

(Please ignore the `TODO` marker above.  This is a note for myself to avoid a flaky time zone-based test.)

**Comment by binkley** [comment]
Related: Consider the Kotlin multiplatform datetime library rather than a custom `Timestamp` class: https://github.com/Kotlin/kotlinx-datetime

**Comment by mjstrasser** [comment]
Thanks @binkley again!

I didn’t know that `kotlinx.datetime` was this advanced. Last time I looked it was very limited. I have removed my `Timestamp` class and delegated timestamp handling to `kotlinx.datetime.Instant`.

Doing this also means I don’t need to provide alternative formatting for timestamps myself. There are no immediate plans to include formatting options in configuration (like Log4j/Logback pattern layouts).
